### PR TITLE
WalletApi: publicMultisigSignerKey method

### PR DESCRIPTION
**Pull request by naughtyfox**
Multisig GUI wallet needs to know it's signer key

### CODE REVIEW

**Comment by naughtyfox** [comment]
> This should call get_multisig_signer_public_key
Done. Check for exception and return empty key string if it's not multisig wallet (the only probable reason for this function to throw)

> Also it's dangerous when the wallet isn't multisig 
Is checking wallet for being multisig enough to make it safe?

**Comment by moneromooo-monero** [comment]
>Is checking wallet for being multisig enough to make it safe?

Should be,though you don't get the ability to get the key before it gets multisig.

**Comment by naughtyfox** [comment]
> Should be,though you don't get the ability to get the key before it gets multisig

That's right. Before wallet became multisig this method will return empty string. I don't consider this situation as an error so don't set error string, i think returning empty string is enough since it's not ambiguous.

File: src/wallet/api/wallet2_api.h [hunk 1]
```diff
@@ -408,6 +408,12 @@ struct Wallet
     */
     virtual std::string publicSpendKey() const = 0;
 
+    /*!
+     * \brief publicSignerKey  - returns public signer key
+     * \return                 - public signer key for multisig wallet or spend public for classic wallet
```
Comment (stoffu) [inline src/wallet/api/wallet2_api.h]: >  or spend public for classic wallet

Wait, this description doesn't agree with the implementation and should be dropped.

How about renaming the function to something like `publicMultisigSignerKey()` to emphasize that it's specific to multisig?

Comment (naughtyfox) [reply src/wallet/api/wallet2_api.h]: no problem!

Comment (naughtyfox) [reply src/wallet/api/wallet2_api.h]: Renamed method and updated descritption

**Pull request merged**